server/ladder_service.py

```diff
@@ -125,8 +131,22 @@ async def fetch_map_pools(self, conn) -> Dict[int, Tuple[str, List[Map]]]:
             _, map_list = map_pool_maps[id_]
             if row.map_id is not None:
                 map_list.append(
-                    Map(row.map_id, row.display_name, row.filename)
+                    Map(row.map_id, row.display_name, row.filename, row.weight)
                 )
+            elif row.other_map_params is not None:
+                param_dict = json.loads(row.other_map_params)
+                if "type" in param_dict:
+                    map_type = param_dict["type"]
+                    if map_type == "neroxis":
+                        if "size" in param_dict and "spawns" in param_dict and "version" in param_dict:
+                            map_size_pixels = param_dict["size"]
+                            spawns = param_dict["spawns"]
+                            version = param_dict["version"]
+                            if isinstance(map_size_pixels, int) and isinstance(spawns, int):
+                                gen_id = version + str(spawns) + str(map_size_pixels)
+                                map_list.append(
+                                    NeroxisGeneratedMap(gen_id, version, spawns, map_size_pixels, row.weight)
+                                )
```

Comment:
10 levels of indentation is a little too much IMO. I would say don't do all this error checking and instead wrap the whole thing in a try block
```suggestion
                try:
                    params = json.loads(row.other_map_params)
                    map_type = params["type"]
                    if map_type != "neroxis":
                        raise Exception("Unsupported map type")
                    map_size_pixels = param_dict["size"]
                    spawns = param_dict["spawns"]
                    version = param_dict["version"]
                    gen_id = version + str(spawns) + str(map_size_pixels)
                    map_list.append(
                        NeroxisGeneratedMap(gen_id, version, spawns, map_size_pixels, row.weight)
                    )
                except Exception:
                    self.logger.warning("Failed to load map [other relevant debug info]", exc_info=True)
```

Reply:
heh yeah that would make it easier

Reply:
although I am a little concerned about the type checking for the spawns and map_size_pixels as this way it wouldn't error out until the get_map was called

Reply:
Well you can check it and just raise an exception like I did with the map type.

Reply:
got it